Libs/MRML/Core/vtkMRMLSliceNode.cxx

```diff
@@ -858,62 +858,31 @@ void vtkMRMLSliceNode::WriteXML(ostream& of, int nIndent)
 
-  of << " fieldOfView=\"" <<
-        this->FieldOfView[0] << " " <<
-        this->FieldOfView[1] << " " <<
-        this->FieldOfView[2] << "\"";
-
-  of << " dimensions=\"" <<
-        this->Dimensions[0] << " " <<
-        this->Dimensions[1] << " " <<
-        this->Dimensions[2] << "\"";
-
-  of << " xyzOrigin=\"" <<
-        this->XYZOrigin[0] << " " <<
-        this->XYZOrigin[1] << " " <<
-        this->XYZOrigin[2] << "\"";
-
-  of << " sliceResolutionMode=\"" << this->SliceResolutionMode << "\"";
-
-  of << " uvwExtents=\"" <<
-        this->UVWExtents[0] << " " <<
-        this->UVWExtents[1] << " " <<
-        this->UVWExtents[2] << "\"";
-
-  of << " uvwDimensions=\"" <<
-        this->UVWDimensions[0] << " " <<
-        this->UVWDimensions[1] << " " <<
-        this->UVWDimensions[2] << "\"";
-
-  of << " uvwOrigin=\"" <<
-        this->UVWOrigin[0] << " " <<
-        this->UVWOrigin[1] << " " <<
-        this->UVWOrigin[2] << "\"";
-
-
-  of << " activeSlice=\"" << this->ActiveSlice << "\"";
-
-  of << " layoutGridRows=\"" <<
-        this->LayoutGridRows << "\"";
-
-  of << " layoutGridColumns=\"" <<
-        this->LayoutGridColumns << "\"";
-
-  std::stringstream ss;
-  int j;
-  for (i=0; i<4; i++)
-    {
-    for (j=0; j<4; j++)
-      {
-      ss << this->SliceToRAS->GetElement(i,j);
-      if ( !( i==3 && j==3) )
-        {
-        ss << " ";
-        }
-      }
-    }
-  of << " sliceToRAS=\"" << ss.str().c_str() << "\"";
-
+  vtkMRMLWriteXMLBeginMacro(of);
+  vtkMRMLWriteXMLStdFloatVectorMacro(fieldOfView, FieldOfView, double);
+  vtkMRMLWriteXMLStdIntVectorMacro(dimensions, Dimensions, int);
+  vtkMRMLWriteXMLStdFloatVectorMacro(xyzOrigin, XYZOrigin, double);
+  vtkMRMLWriteXMLStringMacro(sliceResolutionMode, SliceResolutionMode);
+  vtkMRMLWriteXMLStdFloatVectorMacro(uvwExtents, UVWExtents, double);
+  vtkMRMLWriteXMLStdIntVectorMacro(uvwDimensions, UVWDimensions, int);
+  vtkMRMLWriteXMLStdFloatVectorMacro(uvwDimensions, UVWOrigin, double);
+  vtkMRMLWriteXMLIntMacro(activeSlice, ActiveSlice);
+  vtkMRMLWriteXMLStringMacro(layoutGridRows, LayoutGridRows);
+  vtkMRMLWriteXMLStringMacro(layoutGridColumns, LayoutGridColumns);
+  vtkMRMLWriteXMLMatrix4x4Macro(sliceToRAS, SliceToRAS);
+  vtkMRMLWriteXMLStdStringMacro(orientation, Orientation);
+  vtkMRMLWriteXMLStringMacro(defaultOrientation, DefaultOrientation);
+  vtkMRMLWriteXMLStringMacro(orientationReference, OrientationReference);
+  vtkMRMLWriteXMLIntMacro(jumpMode, JumpMode);
+  vtkMRMLWriteXMLIntMacro(sliceVisibility, SliceVisible);
+  vtkMRMLWriteXMLIntMacro(widgetVisibility, WidgetVisible);
+  vtkMRMLWriteXMLIntMacro(widgetVisibility, WidgetOutlineVisible);
+  vtkMRMLWriteXMLIntMacro(useLabelOutline, UseLabelOutline);
+  vtkMRMLWriteXMLIntMacro(sliceSpacingMode, SliceSpacingMode);
+  vtkMRMLWriteXMLStdFloatVectorMacro(prescribedSliceSpacing, PrescribedSliceSpacing, double);
+  vtkMRMLWriteXMLStdStringVectorMacro(threeDViewNodeRef, ThreeDViewIDs, std::string);
```

Comment:
To ensure the MRML scene are identically serialized, this should be moved after `orientationMatrix`